### PR TITLE
Add support for leveled logging. Add key log statements

### PR DESCRIPTION
**Pull request by vgarg**
The change imports the logrus leveled logger to logrus logger into the ec2 and aws packages. It also introduces log statements at key locations to enable request and response tracking.

@wes-r please review.

### CODE REVIEW

File: aws/client.go [hunk 1]
```diff
@@ -142,9 +151,11 @@ func awsRetry(req *http.Request, res *http.Response, err error) bool {
 		body, _ := ioutil.ReadAll(res.Body)                // Read the body
 		res.Body = ioutil.NopCloser(bytes.NewReader(body)) // Restore the reader
 		if int64(len(body)) != res.ContentLength {
+			dump, _ := httputil.DumpResponse(res, true)
+			log.Warnf("Retryable error. Content length mismatch.\n%s %s\n%v",
```
Comment (wes-r) [inline aws/client.go]: Maybe print out the lengths that were mismatched.

File: ec2/ec2.go [hunk 2]
```diff
@@ -138,20 +139,21 @@ func (ec2 *EC2) query(params map[string]string, resp interface{}) error {
...
 
-	if debug {
-		dump, _ := httputil.DumpResponse(r, true)
-		log.Printf("response:\n")
-		log.Printf("%v\n}\n", string(dump))
+	dump := []byte{}
+	if tracingEnabled() {
+		dump, _ = httputil.DumpResponse(r, true)
+	} else {
+		dump, _ = httputil.DumpResponse(r, false)
```
Comment (wes-r) [inline ec2/ec2.go]: As appose to before, this will always get called if you are not debugging or tracing. Is this line expensive?

**Comment by wes-r** [comment]
LGTM

File: aws/client.go (hunk 1 above)
Comment (vgarg) [inline aws/client.go]: Added

File: ec2/ec2.go (hunk 2 above)
Comment (vgarg) [inline ec2/ec2.go]: Good point. Ideally we would be able to wrap this whole thing into if log.isDebugEnabled(). Haven't found a way to do that with logrus yet.

Just printing the response headers is not that expensive. On the other hand printing body can be quite expensive depending on the body. But we will only be printing the body when we know we are doing debugging using TRACE

**Pull request merged**